### DISCUSSION TITLE
different kernel structure for each latent function in LMCVariationalStrategy

### DISCUSSION THREAD

**Discussion by michaelcao28** [cornellius-gp/gpytorch#2232] [Q&A] [answered]
Hi,

Suppose I am creating an LMC multi-task GP with num_task = num_latents = 3. How do I enable each latent function to have a different kernel structure? For example, latent_1 to have RBFKernel, latent_2 has PeriodicKernel, and latent_3 has MaternKernel. 

Thanks!

**Comment by gpleiss** [comment, marked as answer]
Are you doing exact regression, or are you using a variational model? If you're doing exact regression, you can use the LCMKernel: https://docs.gpytorch.ai/en/stable/kernels.html#gpytorch.kernels.LCMKernel

**Reply by heikestein** [reply]
Hi Geoff, 
I'm trying to do something similar, with multiple kernels of the same type that have distinct active input dimensions (and that should have a fixed but distinct mapping on output dimensions). 
I have a large dataset and am using a variational model. Is there a way to do this? 
Thanks!

**Reply by heikestein** [reply]
Ok, if anyone's reading along, I think I've solved the issue by passing on a `num_latents x num_active_dim` shaped tensor of `kernel.active_dims` indices to the `LazyEvaluatedKernelTensor`'s `_get_item()` method, and then indexing into different input dimensions of inputs `x1` and `x2` after expanding them to `batch_shape`.  
Regarding fixed output mappings, this is easily solved by fixing the `lmc_coefficients` of the `LMCVariationalStrategy` to the respective values.
For now this seems to work as intended, so I consider this issue solved.

**Reply by fjaraavilaa** [reply]
Hey, it is still not super clear to me. In the end you used the LMC Variational Strategy and also the LMC kernel? I am going on with a similar issue.
